ext/crates/fp/src/vector_inner.rs

```diff
@@ -488,10 +485,16 @@ impl<'a, const P: u32> SliceP<'a, P> {
     }
 
     #[inline]
-    fn limb_range(&self) -> (usize, usize) {
+    fn limb_range(&self) -> Range<usize> {
         limb::range::<P>(self.start, self.end)
     }
 
+    #[inline]
+    fn limb_range_inner(&self) -> Range<usize> {
+        let range = self.limb_range();
+        (range.start + 1)..(range.end - 1)
```

Comment:
You're right. I think using `saturating_sub` would be enough to prevent that, while staying correct. This only happens if we're taking a slice that begins and ends at 0.

Reply:
I think in either case we want to be clear about what the return value satisfies. Should it always return a valid range, where the end is >= the start?